keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -843,24 +843,40 @@ examples:
 
 ### Termination of containers
 
-In alpha sidecar containers will be terminated as regular containers. No special
+In Alpha sidecar containers will be terminated as regular containers. No special
 or additional signals will be supported.
 
-We will collect feedback from alpha implementation and plan to improve
-termination in Beta. When Pods with sidecars are terminated:
+In Beta we have thought about introducing additional termination grace period fields
+to manage termination duration
+([draft proposal](https://docs.google.com/document/d/1B01EdgWJAfkT3l6CIwNwoskQ71eyuwet3mjiQrMQbU8))
+and leverage these fields to add reverse order termination of sidecar containers
+after the primary containers terminate.
 
+However, we decided on an alternative that don't require additional fields or hooks while keeping
+the desired behaviors when Pods with sidecars are terminated:
+1. Sidecar containers that have a `PreStop` hook will be notified when the Pod is terminating
+   by executing the `PreStop` hook. This happens at the same time as regular containers, and begins
+   the Pod's termination grace period countdown.
+2. Sidecar containers enter the `Terminated` state and are no longer restarted if they fail.
+3. Once the last primary container terminates, the last started sidecar container is notified by
+   sending a `SIGTERM` signal.
+4. The next sidecar (in reverse order) is notified by sending a `SIGTERM` signal after the previous
+   sidecar container terminates.
+5. This continues until all sidecar containers have terminated, or the Pod's termination grace period
+   expires. In the latter case, the Pod is force terminated and all sidecar containers are killed by
+   sending a `SIGKILL` signal.
```

Comment:
This looks little bit confusing. A container will always get minimum grace period (2s) even if the termination grace period expires.

https://github.com/kubernetes/kubernetes/pull/120620

Reply:
nit:

'In the latter case, all containers are terminated with minimum grace period and the Pod will be terminated after that.'